rakefile

```diff
@@ -1,13 +1,85 @@
 require 'rototiller'
 require 'rspec/core/rake_task'
+require 'net/http'
+require 'timeout'
+require 'json'
+
+BASE_ABS_URI = 'https://cinext-abs.delivery.puppetlabs.net/api/v2/'
 
 task :default => :performance
 
-desc 'Run Performance setup'
-rototiller_task :performance do |t|
+desc 'Provision systems using ABS (or use already set ABS_RESOURCE_HOSTS) and then execute beaker performance setup and tests'
+rototiller_task :performance do
+  Rake::Task["performance_provision_with_abs"].execute unless ENV['ABS_RESOURCE_HOSTS']
+  Rake::Task["performance_without_provision"].execute
+  Rake::Task["performance_deprovision_with_abs"].execute unless ENV['BEAKER_PRESERVE_HOSTS'] == 'always' ||
+      ((@beaker_cmd.nil? || @beaker_cmd.result.exit_code != 0)&&
+          ENV['BEAKER_PRESERVE_HOSTS'] == 'onfail')
+end
+
+def retry_abs_request(uri, body, timeout, time_increment, invalid_response_bodies = [], valid_response_codes = [202, 200])
+  req = Net::HTTP::Post.new(uri, 'Content-Type' => 'application/json')
+  req.body = body
+  http = Net::HTTP.new(uri.host, uri.port)
+  http.use_ssl = true
+
+  puts "sending request body: #{body}"
+  puts "to uri: #{uri}"
+
+  res = nil
+  Timeout::timeout(timeout) do
+    while res.nil? || invalid_response_bodies.include?(res.body) || !valid_response_codes.include?(res.code)
+      begin
+        res = http.request(req)
+      rescue => ex
+        puts ex
+        puts ex.backtrace
+      end
+      if res.nil? || invalid_response_bodies.include?(res.body)
+        puts "#{Time.now} - Retrying ABS request..."
+        sleep(time_increment)
+      else
+        break
+      end
+    end
+  end
+  res
+end
+
+desc 'Provision systems for the performance task with ABS - included in the performance task'
+rototiller_task :performance_provision_with_abs do
```

Comment:
these tasks should probably be organized by namespace later...
something like `performance:provision_abs` etc

but i'm fine deferring UX changes like this until later